packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -240,4 +251,109 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
       LIMIT 1;
     `);
   }
+
+  async getUsersWhoHaveMadeDialogues(): Promise<DbUser[]> {
+    return this.getRawDb().any(`
+      WITH all_dialogue_authors AS
+        (SELECT (UNNESTED->>'userId') AS _id
+            FROM "Posts" p, UNNEST("coauthorStatuses") unnested
+            WHERE p."collabEditorDialogue" IS TRUE 
+            AND p."draft" IS FALSE
+        UNION
+        SELECT p."userId" as _id
+            FROM "Posts" p
+            WHERE p."collabEditorDialogue" IS TRUE
+            AND p."draft" IS FALSE
+        )
+      SELECT u.*
+      FROM "Users" u
+      INNER JOIN all_dialogue_authors ON all_dialogue_authors._id = u._id
```

Comment:
If you're just using this to get the list of user ids, this can just be:
```
WHERE u._id IN (
  SELECT * FROM all_dialogue_authors
)
```

(I personally find this more comprehensible than a join, though Ruby says that this isn't idiomatic.  Not a string push to change it.)